File: C#/0559. Maximum Depth of N-ary Tree.cs
/*
// Definition for a Node.
public class Node {
    public int val;
    public IList<Node> children;

    public Node(){}
    public Node(int _val,IList<Node> _children) {
        val = _val;
        children = _children;
}
*/
public class Solution {
    public int MaxDepth(Node root) {
        if(root==null){
            return 0;
        }
        if (root.children.Count==0){
            return 1;
        }
        int rep=0;
        foreach(Node child in root.children){
            rep=Math.Max(repa,1+MaxDepth(child));
        }
        return rep;
    }
}

File: 20190519python/0559. Maximum Depth of N-ary Tree.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val, children):
        self.val = val
        self.children = children
"""
class Solution:
    def maxDepth(self, root: 'Node') -> int:
        if not root:
            return 0
        root_list=[root]
        cnt=0
        while root_list:
            cnt+=1
            cur_root_count=len(root_list)
            for i in range(cur_root_count):
                cur_root=root_list.pop(0)
                for child in cur_root.children:
                    root_list.append(child)
        return cnt
